integration_tests/README.md

```diff
@@ -5,41 +5,161 @@ are intended to be able to be run against any Spark-compatible cluster/release t
+</p>
+</details>
+
+<details><summary><b>Installing specific version of maven</b></summary>
+<p>
+
+All package managers like `brew` and `apt` offers maven. However, it may lag behind some
+versions. In that case, you can install the latest binary from the [Maven download page](https://maven.apache.org/download.cgi).
+For manual installation, you need to setup your environment:
+
+  ```shell script
+  export M2_HOME=PATH_TO_MAVEN_ROOT_DIRECTOTY
+  export M2=${M2_HOME}/bin
+  export PATH=$M2:$PATH
+  ```
+
+</p>
+</details>
+
+### Dependencies
+
+- pytest
+  : A framework that makes it easy to write small, readable tests, and can scale to support complex
+  functional testing for applications and libraries (requires  Python 3.6+).
+- sre_yield
+  : Provides a set of APIs to generate string data from a regular expression.
+- pandas
+  : A fast, powerful, flexible and easy to use open source data analysis and manipulation
+  tool and is only needed when testing integration with pandas.
+- pyarrow
+  : Provides a Python API for functionality provided by the Arrow C++ libraries, along with
+  tools for Arrow integration and interoperability with pandas, NumPy, and other software in
+  the Python ecosystem. This is used to test improved transfer performance to pandas based user
+  defined functions.
+- pytest-xdist
+  : A plugin that extends pytest with new test execution modes, the most used being distributing
+  tests across multiple CPUs to speed up test execution
+- findspark
+  : Adds pyspark to sys.path at runtime
+
+You can install all the dependencies using `pip` by running the following command:
+
+  ```shell script
+  pip install pytest \
+              sre_yield \
+              pandas \
+              pyarrow \
+              pytest-xdist \
+              findspark
+  ```
+
+### Installing Spark
+
+You need to install spark-3.x and set `SPARK_HOME/bin` to your `$PATH`.
+
+To build a distribution run the following command. Full instructions can be found [here](https://spark.apache.org/docs/latest/building-spark.html).
+
+To run the plugin against the latest spark, you will need to buid a new version from source:
+
+  ```shell script
+  ## to build a distribution from branch is branch-3.x.y
+  SPARK_BRANCH=branch-3.x.y
+  SPARK_VER=3xy
+  
+  git clone -b ${SPARK_BRANCH} https://github.com/apache/spark.git spark-src-${SPARK_VER}
+  cd spark-src-${SPARK_VER}
+  ## build a distribution with hive support
+  ./dev/make-distribution.sh --name spark-for-rapids-${SPARK_VER} --tgz -Pkubernetes -Phive
+  ## this will generate a single tgz file that you would need to untar it to $SPARK_INSTALLS_DIR
+  ```
+
+Set the environment:
+
+  ```shell script
+  SPARK_HOME=$SPARK_INSTALLS_DIR/spark-${SPARK_VER}
+  export SPARK_HOME
+  export PATH=${SPARK_HOME}/bin:$PATH
+  ```
+
+### Building The Plugin
+
+Make sure that you compile the plugin against the same version of Spark that it is going to run with.
```

Comment:
This should probably just link to the contributor's guide which covers how to build the plugin.  Putting a copy of instructions here is likely to decay into incorrect instructions over time.